src/workspaces/features/DeleteDatabaseTable/DeleteDatabaseTableDialog.tsx

```diff
@@ -0,0 +1,96 @@
+import { gql } from "@apollo/client";
+
+import { Trans, useTranslation } from "react-i18next";
+import { useRouter } from "next/router";
+import Button from "core/components/Button";
+import Spinner from "core/components/Spinner";
+import useCacheKey from "core/hooks/useCacheKey";
+import { DeletePipelineVersionError, PipelineError } from "graphql-types";
+import { useState, version } from "react";
+import {
+  useDeletePipelineMutation,
+  useDeletePipelineVersionMutation,
+  useDeleteWorkspaceDatabaseTableMutation,
+} from "workspaces/graphql/mutations.generated";
+import Dialog from "core/components/Dialog";
+import {
+  DatabaseTableDelete_DatabaseFragment,
+  DatabaseTableDelete_WorkspaceFragment,
+} from "./DeleteDatabaseTableDialog.generated";
+
+type DeletePipelineDialogProps = {
+  open: boolean;
+  onClose: () => void;
+  table: DatabaseTableDelete_DatabaseFragment;
+  workspace: DatabaseTableDelete_WorkspaceFragment;
+};
+
+const DeleteDatabaseTableDialog = (props: DeletePipelineDialogProps) => {
```

Comment:
Can you just use a trigger like component ? It's less code and more implemented for this kind of use case.